src/test/java/se/liu/ida/hefquin/engine/federation/catalog/FederationDescriptionReaderTest.java

```diff
@@ -56,4 +57,109 @@ public void twoFMs() {
 		assertTrue( fm2.getInterface() instanceof TPFInterface );
 	}
 
+	@Test
+	public void sparqlFMWithTwoEndpoints() {
+		final String turtle =
+				"PREFIX fd:     <http://www.example.org/se/liu/ida/hefquin/fd#>\n"
+						+ "PREFIX ex:     <http://example.org/>\n"
+						+ "\n"
+						+ "ex:dbpediaSPARQL\n"
+						+ "      a            fd:FederationMember ;\n"
+						+ "      fd:interface [ a                  fd:SPARQLEndpointInterface ;\n"
+						+ "                     fd:endpointAddress <http://dbpedia.org/sparql>, <http://localhost:7474/db/neo4j/tx> ];\n"
+						+ "                     fd:vocabularyMappingsFile \"dbpedia/vocabularyMappings.nt\".";
+
+		final Model fd = ModelFactory.createDefaultModel();
+
+		final RDFParserBuilder b = RDFParser.fromString(turtle);
+		b.lang( Lang.TURTLE );
+		b.parse(fd);
+
+		Exception exception = assertThrows(IllegalArgumentException.class, () -> {
+			FederationDescriptionReader.readFromModel(fd);
+		});
+		// Test that the error message is correct
+		String expectedErrorMessage = "More Than One EndpointAddress!";
+		String actualErrorMessage = exception.getMessage();
```

Comment:
Please always use the `final` keyword for variables that are not meant to be reassigned.
```suggestion
		final Exception exception = assertThrows(IllegalArgumentException.class, () -> {
			FederationDescriptionReader.readFromModel(fd);
		});
		// Test that the error message is correct
		final String expectedErrorMessage = "More Than One EndpointAddress!";
		final String actualErrorMessage = exception.getMessage();
```

Reply:
I changed all variables to final in the test file.